File: FitBook.Model/Requests/NewsItems/NewsItemInsertRequest.cs
namespace FitBook.Model.Requests.NewsItems;

public class NewsItemInsertRequest
{
    public string Title { get; set; } = string.Empty;
    public string Content { get; set; } = string.Empty;
    public string ImageUrl { get; set; } = string.Empty;
    public bool IsActive { get; set; }
}

File: UI/fitbook_desktop/lib/models/requests/news_item_insert_request.dart
import 'package:json_annotation/json_annotation.dart';

import '../common/api_request_body.dart';

part 'news_item_insert_request.g.dart';

/// Mirrors `FitBook.Model.Requests.NewsItemInsertRequest`.
@JsonSerializable()
class NewsItemInsertRequest implements ApiRequestBody {
  NewsItemInsertRequest({
    required this.title,
    required this.content,
    required this.imageUrl,
    required this.isActive,
  });

  final String title;
  final String content;
  final String imageUrl;
  final bool isActive;

  factory NewsItemInsertRequest.fromJson(Map<String, dynamic> json) =>
      _$NewsItemInsertRequestFromJson(json);

  @override
  Map<String, dynamic> toJson() => _$NewsItemInsertRequestToJson(this);
}
